### DISCUSSION TITLE
FFCV with PyTorch 2.0

### DISCUSSION THREAD

**Discussion by tavisshore** [libffcv/ffcv#284] [Q&A] [answered]
I'm trying to test out the apparent performance improvements of PyTorch 2.0 over 1.13.1 but I'm having trouble installing it with FFCV. 

Has anyone been able to or is it just not compatible yet?

**Comment by richardcepka** [comment]
Are there any updates? I am also interested in this issue.

**Comment by andrewilyas** [comment]
@tavisshore We haven't tested with PyTorch 2.0 yet---what issues are you running into?

**Reply by tavisshore** [reply]
When installing environment requirements, if I run the command shown in [ffcv.io](url), replacing for PyTorch 2.0 there's a compatibility error. From what I can see it's due to the compilers package. I will try again now as PyTorch 2.0 stable has been released

**Reply by andrewilyas** [reply, marked as answer]
Please let me know how it goes! If there are special installation instructions for 2.0, I'd suggest:

1. Make a new conda env
2. Install pytorch 2.0 with conda using whatever the right command is
3. paste in the rest of the install command on ffcv.io (which, fyi, has changed to not include ``compilers'' with the release of 1.0), MINUS the pytorch parts (i.e., no pytorch, cudatoolkit, or torchvision)

Let me know how it goes!

**Reply by SerezD** [reply]
Hi, I installed ffcv with pytorch 2.0.0 and python 3.10 (no need for compilers on my environment)

Here is the full process:

```
conda create --name ffcv_env python=3.10
conda activate ffcv_env

# Install torch
conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia

# install dependences
conda install cupy pkg-config libjpeg-turbo opencv numba  -c pytorch -c conda-forge

# install ffcv
pip install ffcv
```

Unfortunately, the same thing does not work with python 3.11 on my environment, due to conflicts with `cupy`

**Reply by tavisshore** [reply]
Thanks - sorry I forgot to update the thread, removing the compilers package from requirements solved this issue
